### PR TITLE
Restructuring project

### PR DESCRIPTION
**Pull request by griffobeid**
* Moved dev docker-compose to root
* Changed server folder -> nodejs
* Changed server_rust folder -> rust

I tested rust and nodejs locally

### CODE REVIEW

File: server/package-lock.json [hunk 1]
```diff
@@ -1,12109 +0,0 @@
-{
```
Comment (griffobeid) [inline server/package-lock.json:1]: Deleting this file in favor of `yarn.lock`

File: rust/docker-compose.yaml [hunk 2]
```diff
@@ -1,17 +1,12 @@
-version: "3.4"
```
Comment (JasterV) [inline rust/docker-compose.yaml:1]: I was getting an error because version 3.0 does not support the cache_from property, I'm not sure if it will work now

Comment (griffobeid) [reply rust/docker-compose.yaml:1]: 3.0 is old but 3 will use the latest version in the major release of "3.X"

Comment (JasterV) [reply rust/docker-compose.yaml:1]: yay okey

**Review by JasterV — APPROVED** [review]

**Pull request merged**